system-addon/lib/SectionsManager.jsm

```diff
@@ -151,21 +151,37 @@ const SectionsManager = {
     }
   },
 
+  /**
+   * Save Top Story metadata to places db.
+   */
   updateBookmarkMetadata({url}) {
-    this.sections.forEach(section => {
-      if (section.rows) {
-        section.rows.forEach(card => {
-          if (card.url === url && card.description && card.title && card.image) {
-            PlacesUtils.history.update({
-              url: card.url,
-              title: card.title,
-              description: card.description,
-              previewImageURL: card.image
-            });
-          }
-        });
-      }
-    });
+    const {rows} = this.sections.get("topstories");
```

Comment:
Similar to the another comment about being specific to top stories or general for any section… if it's story specific, this should probably be in `TopStoriesFeed`. Or if it's supposed to be more general, e.g., for webextension, then maybe it should check each section.. except highlights? @k88hudson any thoughts on if we should be saving metadata when bookmarking any section's cards?

Reply:
It would probably be a good idea to save metadata if we have it, yeah 👍

Reply:
@Mardak I think building something general enough seems out of scope for this bug since webextensions are somewhat experimental. Maybe for now the logic should be handled by the TopStoriesFeed since its very specific.

Reply:
The general approach is what the code was doing before. Except it should skip highlights to avoid the bug you noticed.